### ISSUE TITLE
Relations betwen fields

### ISSUE THREAD

**Issue by A11oW** [jquense/react-formal#79]
How I can validate two fields with relation?
For example I have time line with `start` and `end` fields, where `end` must be larger `start`.

**Comment by maullerz** [comment]
Hello!
You can relate on other field like this:
```
  // Period
  startDate: yup.date(),
  endDate: yup.date().min(yup.ref('startDate'), 'endDate cannot be earlier than startDate!'),
```

**Comment by A11oW** [comment]
I tried this variant, but then object errors always contains `endDate cannot be earlier than startDate!`. I think endDate must use `.test` or `.when`, but I don't know how.
I found this issue https://github.com/jquense/yup/issues/33, but this don't work for me.

**Comment by jquense** [comment]
```js
var yup = require('yup')

var schema = yup.object({
  start: yup.date(),
  end: yup.date().min(yup.ref('start'), 'End date must be after the start date')
})

await schema.validate({
  start: new Date(2016, 5, 6),
  end: new Date(2016, 5, 7),
})
```

The above example is working fine for me.

**Comment by A11oW** [comment]
This schema working good in yours case through validate method, but if I use he with react-formal `Field.Message` always displayed the error for 'end' field when field changed. But `Form.Button` validate Form correct.

**Comment by jquense** [comment]
I'd have to see your isolated form, and default values, it may not handle null well. you also probably want to validate those fields together with alsoValidates to on the field

**Comment by A11oW** [comment]
I didn't do anything special. I create [gist](https://gist.github.com/A11oW/42b9d9ac25b44a19dce35ec656b6ac66) with code. If needed I will create form in any sandbox.

**Comment by jquense** [comment]
Awesome thanks for the runnable example it helped me find a bug in react-formal here!

**Comment by jquense** [comment]
fixed in 20.3

**Comment by A11oW** [comment]
After update all works good. Thanks for fast update!
